### ISSUE TITLE
[BUG] lazy loading regression

### ISSUE THREAD

**Issue by lmeyerov** [graphistry/pygraphistry#481]
after 0.29.0 (big pyg[ai] merge), looks like import is slow again, likely some top-level imports getting triggered:

![Screenshot from 2023-05-02 21-24-27](https://user-images.githubusercontent.com/4249447/235832882-abfa5234-4bc8-47ce-8bb0-1346e8f9efe0.png)

for reference, pandas is just 600ms on same box -- so we can be targeting subsecond, but aren't

**Comment by lmeyerov** [comment]
as part of the fix, maybe we add a unit test? at least '1 out of 3 cold starts should be subsecond'

**Comment by silkspace** [comment]
@tanmoyio could this have to do with new inits?

**Comment by lmeyerov** [comment]
@tanmoyio @dcolinmorgan  can we look as part of landing cu_cat?

**Comment by lmeyerov** [comment]
I documented how I tracked it down last time: https://www.graphistry.com/blog/import-pygraphistry-from-10s-to-1s-by-python-import-lazy-loading-and-tuna

**Comment by dcolinmorgan** [comment]
as in lazy loading cucat?
[import.orig.log](https://github.com/graphistry/pygraphistry/files/12143398/import.orig.log)
<img width="1058" alt="Screenshot 2023-07-24 at 16 17 45" src="https://github.com/graphistry/pygraphistry/assets/1395951/af7dd49e-e852-42a3-84da-9b49fef5cdb5">

**Comment by lmeyerov** [comment]
i mean why is `import graphistry` slow -- is it pulling in some unexpected deps during import time that need to be lazy loaded?

**Comment by lmeyerov** [comment]
w/ lazy loading, `cu_cat`, `cudf`, etc shouldn't load at `import graphistry`, i'd expect at most just `pandas`

**Comment by dcolinmorgan** [comment]
yes i see the issue now on 29.3 -- on eks where cuda is available, cudf loads in 4.7s for total g. load time of 6.7s, whereas on my mac laptop entire graphistry 29.3 loads in 0.829s. seems lazy cudf is getting initiated before user request. So the question is to front load or break loadings into pieces.

re: #439 seems like there are 2 options: 

1. if cudf is available, load it and run by default in 'auto' flag. otherwise
2. wait to load cudf until user triggers with cu-cat/cuml flags
<img width="1228" alt="Screenshot 2023-07-27 at 18 54 48" src="https://github.com/graphistry/pygraphistry/assets/1395951/77c0724a-d004-4bdc-8542-65a339c50e98">

**Comment by lmeyerov** [comment]
Option 3. Lazy load. Delay any cudf imports until the code using it (including auto) is run. 'import graphistry' shouldn't need to import cudf.

**Comment by dcolinmorgan** [comment]
ok yeah i see the issue in `embed_utils` -- at some point we needed to check between cudf. and pd. df and loaded cudf to do that, so i just swapped to using `getmodule` and loadtime back down to 1.8s all on pandas
